homeassistant/components/mobile_app/__init__.py

```diff
@@ -103,12 +104,15 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     registration_name = f"Mobile App: {registration[ATTR_DEVICE_NAME]}"
     webhook_register(hass, DOMAIN, registration_name, webhook_id, handle_webhook)
 
+    cloud_hook_lock = asyncio.Lock()
+
     async def create_cloud_hook() -> None:
         """Create a cloud hook."""
-        hook = await cloud.async_create_cloudhook(hass, webhook_id)
-        hass.config_entries.async_update_entry(
-            entry, data={**entry.data, CONF_CLOUDHOOK_URL: hook}
-        )
+        async with cloud_hook_lock:
+            hook = await cloud.async_get_or_create_cloudhook(hass, webhook_id)
```

Comment:
Like we discussed earlier I think it would be good to have a single place for all the cloud hook code in the mobile app integration. There are now multiple places that do things with cloud hooks in the integration and since there needs to be coordination and synchronization it's hard to reason about it.

Reply:
Done